src/ai/defaultai.cc

```diff
@@ -5648,8 +5647,8 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 			inputs[9] = (persistent_data->trees_around_cutters < 200) * 1;
 			inputs[10] = (persistent_data->trees_around_cutters < 300) * 1;
 			inputs[11] = (persistent_data->trees_around_cutters < 400) * 1;
-			inputs[12] = (wood_policy_ != WoodPolicy::kAllowRangers) * 1;
-			inputs[13] = (wood_policy_ != WoodPolicy::kAllowRangers) * 1;
+			inputs[12] = (wood_policy_ == WoodPolicy::kAllowRangers) * 1;
```

Comment:
I think `(wood_policy_ == WoodPolicy::kAllowRangers)  ? 1 : 0;` wozuld make things more clear.
(Mixing boolen and int is an inheritance of C, well)

Reply:
I could do this as it is done like this in other parts of the code. One short question as I reall don't know it: might there be a difference in performance between the 2 variants

Reply:
Performance impact should be negligible with a modern compiler, but readabily will improve a lot ...